src/sentry/auth/provider.py

```diff
@@ -17,6 +17,13 @@ class MigratingIdentityId(namedtuple("MigratingIdentityId", ["id", "legacy_id"])
     MigratingIdentityId may be used in the ``id`` field of an identity
     dictionary to facilitate migrating user identities from one identifying id
     to another.
+
+    Context - when google oauth was initially created, the auth_identity key was simply
+    the provider email. This can cause issues if the customer changes their domain name,
+    and now their email is different and they're locked out of their account.
+    This logic updates their id to the provider id instead.
+
+    NOTE: this should _only_ really be relevant for google oauth implementation
```

Comment:
In the future if we needed to migrate other identities we would probably use this as well. But we've never had to do that

Reply:
ya. this is kinda specific though.
At this point it owuld be if we ever needed to migrate away from 'user_id' which... we likely shouldn't use anything other than the provider user_id anyways

Reply:
Well, this isn't about migrating away from `user_id`, it's about migrating away from a specific field from the oAuth response of the provider to another response from the provider.

In google's case it was that we migrated from email → sub. It could happen to other providers as well if for some reason we picked the wrong unique identifier